### ISSUE TITLE
✨ Community UI for viewing people who have expressed interest

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#12290]
## ✨ Feature

For people who have added a community to their profile AND selected that they are interested in job opportunities (or training opportunities) with that community:
Make their profiles available to recruiters in the community.

## 🕵️ Details

New page URL: talent.canada.ca/en/admin/community-talent
Breadcrumb: Home > Community dashboard > Community talent

## 🎨 Design

[Figma file](https://www.figma.com/design/uZSzaV8ygi2SA1LwQlBIhh/Unassessed-Candidates?node-id=88-16198&t=mQ9L6ksyzQeMCvBh-1) by @Jerryescandon 

## 📸 Screenshot

![Image](https://github.com/user-attachments/assets/6627d243-e472-4236-b76f-b09c80185c9c)

## 🙋‍♀️ Proposed Implementation

We thought we could re-use the User table for this, but that's an old table that doesn't reflect our new admin patterns. So I think we'll just have to create a new table instead.

Actually, if you wanted to update the User table to meet the specs below, then show it in both the User page and this new Community talent page, that would be ideal. But I feel like the rows are different on this new table, since the same user can show multiple times.

## ✅ Acceptance Criteria

- [ ] Create new page
  - [ ] Accessible by users with the Community Recruiter and Community Admin roles only
- [ ] Add table showing community interests from #12603 
   - [ ] Column: Name (links to profile), Community (hidden by default), Classification (current Group-level), Work email (hidden by default), Working language ability, Job interest, Training interest, Preferred communication language (hidden by default)
   - [ ] only name column is a link
   - [ ] Separate row for EACH community interest (you have 2 rows if you expressed interest in 2 different communities)
   - [ ] Default null state well if no rows
   - [ ] Clicking a user's name goes to their profile page, like a pool candidate

> [!NOTE]
>  Out of scope:
> - filter dialog
> - file downloads
> - search

## 🛑 Blockers

```[tasklist]
### Blocked By
- [ ] #12603 
```

**Comment by Jerryescandon** [comment]
I have concerns about how we should be best handling this.

It seems simple to just add a new URL and table to admin/community. Which would allow simple searches from the community side.

But in the context of answering requests. Would it not be better to add this as a filter/column to the current Candidate table? This would allow the names to show up in the request results. Potentially also in the Talent search for manager side.

**Comment by tristan-orourke** [comment]
@Jerryescandon remember that our "Candidate" tables represent `PoolCandidate` objects, which are more like a job application than a person. The person is represented by a `User` object, which is a "parent" of `PoolCandidate`, and (potentially) has many `PoolCandidates`.

When people are expressing interest in a Community, they're doing so as a `User`, without having to apply to any pool.

So, I don't know if we can represent `Users` and `PoolCandidates` in one table at the same time. Or rather, we could, but I don't know if we should. There's already confusion about whether a given page is representing a person or an application, I think we should be trying to make those more distinct, not less.

And if we do it, there will be non-trivial frontend and backend work involved. Sadly, not just the addition of a new filter.
